app/javascript/responseTemplates/responseTemplates.js

```diff
@@ -0,0 +1,249 @@
+/* eslint-disable no-alert */
+/* eslint-disable no-restricted-globals */
+
+export function toggleTemplateTypeButton(form, e) {
+  const { targetType } = e.target.dataset
+  const activeType = targetType === 'personal' ? 'moderator' : 'personal';
+  e.target.classList.toggle('active');
+  form.querySelector(`.${activeType}-template-button`).classList.toggle('active');
+
+  form.querySelector(`.${targetType}-responses-container`).classList.toggle('hidden');
+  form.querySelector(`.${activeType}-responses-container`).classList.toggle('hidden');
+}
+
+function buildHTML(response, typeOf) {
+  if (response.length === 0 && typeOf === 'personal_comment') {
+    return `
+<div class="mod-response-wrapper mod-response-wrapper-empty">
+  <p>🤔... It looks like you don't have any templates yet.</p>
+  <p>Create templates to quickly answer FAQs or store snippets for re-use.</p>
+</div>
+`;
+  }
+  if (typeOf === 'personal_comment') {
+    return response
+      .map((obj) => {
+        return `
+          <div class="mod-response-wrapper">
+            <span>${obj.title}</span>
+            <p>${obj.content}</p>
+            <button class="insert-template-button" type="button" data-content="${obj.content}">INSERT</button>
+          </div>
+        `;
+      })
+      .join('');
+  }
+  if (typeOf === 'mod_comment') {
+    return response.map((obj) => {
+      return `
+            <div class="mod-response-wrapper">
+              <span>${obj.title}</span>
+              <p>${obj.content}</p>
+              <button class="insert-template-button" type="button" data-content="${obj.content}">INSERT</button>
+              <button class="moderator-submit-button" type="submit" data-response-template-id="${obj.id}">SEND AS MOD</button>
+            </div>
+          `;
+    })
+    .join('');
+  }
+  return `Error 😞`;
+}
+
+function submitAsModerator(responseTemplateId, parentId) {
+  const commentableId = document.querySelector('input#comment_commentable_id')
+    .value;
+
+  fetch(`/comments/moderator_create`, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      response_template: {
+        id: responseTemplateId,
+      },
+      comment: {
+        body_markdown: '',
+        commentable_id: commentableId,
+        commentable_type: 'Article',
+        parent_id: parentId,
+      },
+    }),
+  })
+    .then((response) => response.json())
+    .then((response) => {
+      if (response.status === 'created') {
+        window.location = response.path;
```

Comment:
Not entirely sure if this is a security concern... `response.path` comes from the controller's response. I don't think this able to be manipulated but not 100% confident.

If it is a concern I can switch this back to `window.location.reload`.